controllers/auth.js

```diff
@@ -55,21 +55,32 @@ const User = require('../models/User')
   }
   
   exports.postSignup = (req, res, next) => {
-    const validationErrors = []
-    if (!validator.isEmail(req.body.email)) validationErrors.push({ msg: 'Please enter a valid email address.' })
-    if (!validator.isLength(req.body.password, { min: 8 })) validationErrors.push({ msg: 'Password must be at least 8 characters long' })
-    if (req.body.password !== req.body.confirmPassword) validationErrors.push({ msg: 'Passwords do not match' })
-  
+    const validationErrors = [];
+    if (!validator.isEmail(req.body.email))
+        validationErrors.push({ msg: "Please enter a valid email address." });
+    if (!validator.isLength(req.body.password, { min: 8 }))
+        validationErrors.push({
+            msg: "Password must be at least 8 characters long",
+        });
+    if (req.body.password !== req.body.confirmPassword)
+        validationErrors.push({ msg: "Passwords do not match" });  
+    if (req.body.firstName !== req.body.firstName)
+      validationErrors.push({ msg: 'Please enter your Name.' }) 
+    if (req.body.lastName !== req.body.lastName)
+      validationErrors.push({ msg: 'Please enter your last Name.' }) 
     if (validationErrors.length) {
-      req.flash('errors', validationErrors)
-      return res.redirect('../signup')
+        req.flash("errors", validationErrors);
+        return res.redirect("../signup");
     }
+    
     req.body.email = validator.normalizeEmail(req.body.email, { gmail_remove_dots: false })
   
     const user = new User({
       userName: req.body.userName,
       email: req.body.email,
-      password: req.body.password
+      password: req.body.password,
+      firstName: req.body.firstName,
+      lastName: req.body.lastName
     })
```

Comment:
It seems that the other fields `dateofBirth`, `location` and `description` are not being stored in the DB so they are lost after the form is submitted.